comps/cores/mega/gateway.py

```diff
@@ -1053,7 +1053,7 @@ async def handle_request(self, request: Request):
         if "metadata" in result_dict[last_node].keys():
             # from retrieval results
             if decoded_audio_input:
-                metadata = {"audio": decoded_audio_input}
+                metadata["audio"] =  decoded_audio_input
```

Comment:
Oh, this doesn't work because metadata doesn't exist yet. So maybe:
```
if "metadata" in result_dict[last_node].keys():
    metadata = result_dict[last_node]["metadata"]
    if decoded_audio_input:
        metadata["audio"] =  decoded_audio_input    
```